torchinductor/ir.py

```diff
@@ -1365,9 +1381,13 @@ def simplify_reorder_and_tile(self):
                 reduce_size.append(s)
 
         def simplify_and_reorder(x_vars, sizes):
-            sizes, reindex1, prune = _simplify_loops(x_vars, sizes, index_formulas)
+            sizes, reindex1 = self._apply_loop_reordering(
+                x_vars, sizes, memory_addrs, priority_idx
+            )
+            x_vars = reindex1(x_vars)
+            sizes, reindex2, prune = _simplify_loops(x_vars, sizes, index_formulas)
             x_vars = prune(x_vars)
-            sizes, reindex2 = self._apply_loop_reordering(x_vars, sizes, memory_addrs)
+            # sizes, reindex2 = self._apply_loop_reordering(x_vars, sizes, memory_addrs, priority_idx)
             reindex = fuse_reindexing(reindex1, reindex2)
             return sizes, reindex
```

Comment:
I try to `_apply_loop_reordering` to force the order to be the same as the row with priority_idx(like Convolution ouput buf0) before `_simplify_loops` to avoid loop flattened to fewer dimensions. Although this modification could make following case works
```
y = conv(x, w)
z = y.transpose(2,3) + bias
```
It will make many cases in test_torchinductor.py fail. But I do not have good idea how to solve this

Reply:
Solved by adding `same_reordering`